lib/semian/atomic_integer.rb

```diff
@@ -0,0 +1,28 @@
+module Semian
+  class AtomicInteger #:nodoc:
+    attr_accessor :value
+
+    def initialize(_name, _permissions)
+      @value = 0
+    end
+
+    def increment_by(val)
+      self.value += val
+    end
+
+    def increment
+      increment_by(1)
+    end
+
+    def execute_atomically
+      yield if block_given?
+    end
+
+    def shared?
+      false
+    end
```

Comment:
> how would you handle testing SysVAtomicInteger on a system that doesn't support semaphores?

You don't test it. It's what we always did. 

> next unless shared? then it would be fine.

If that is what shared is for, then it's bad naming. 
I encourage you to take a look at the `thread_safe` gem, on how they handle code radically different based on the platform: https://github.com/ruby-concurrency/thread_safe